File: crossecore-java/src/java_/ArrayInitializerBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ArrayInitializerBase 
extends ExpressionImpl implements ArrayInitializer
{
	private com.crossecore.ocl.OrderedSet<Expression> _expressions;
	
	public com.crossecore.ocl.OrderedSet<Expression> getExpressions()
	{
		if(_expressions==null){
			_expressions = new com.crossecore.ocl.OrderedSet<Expression>(Expression.class, this, Java_PackageImpl.ARRAY_INITIALIZER__EXPRESSIONS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.ARRAY_INITIALIZER__EXPRESSIONS);
		}
		return _expressions;
	
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.ARRAYINITIALIZER;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.ARRAY_INITIALIZER__EXPRESSIONS:
				return getExpressions();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.ARRAY_INITIALIZER__EXPRESSIONS:
				getExpressions().clear();
				getExpressions().addAll((java.util.Collection<? extends Expression>) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.ARRAY_INITIALIZER__EXPRESSIONS:
				return _expressions != null && !_expressions.isEmpty();
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/ArrayInitializerBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class ArrayInitializerBase 
	:ExpressionImpl, ArrayInitializer
	{
		private OrderedSet<Expression> _expressions;
		
		public virtual OrderedSet<Expression> expressions
		{
			get {
				if(_expressions==null){
					_expressions = new OrderedSet<Expression>(this, Java_PackageImpl.ARRAYINITIALIZER_EXPRESSIONS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.ARRAYINITIALIZER_EXPRESSIONS);
				}
				return _expressions;
			}
		
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.ARRAYINITIALIZER;
		}
		
		
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.ARRAYINITIALIZER_EXPRESSIONS:
					return expressions;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.ARRAYINITIALIZER_EXPRESSIONS:
					expressions.Clear();
					expressions.AddRange(((List<EObject>)newValue)?.Cast<Expression>());
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.ARRAYINITIALIZER_EXPRESSIONS:
					return _expressions != null && !_expressions.isEmpty();
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
